jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/inputs/Checkbox.java

```diff
@@ -3,16 +3,26 @@
 import com.epam.jdi.light.asserts.generic.HasAssert;
 import com.epam.jdi.light.common.JDIAction;
 import com.epam.jdi.light.elements.base.UIBaseElement;
+import com.epam.jdi.light.elements.common.UIElement;
 import com.epam.jdi.light.elements.interfaces.base.HasCheck;
 import com.epam.jdi.light.elements.interfaces.base.HasClick;
 import com.epam.jdi.light.material.asserts.inputs.CheckboxAssert;
 
 public class Checkbox extends UIBaseElement<CheckboxAssert> implements HasClick, HasCheck,
-    HasAssert<CheckboxAssert> {
+    HasAssert<CheckboxAssert>{
+
+    public UIElement getCheckBoxes() {
+        return this.core();
+    }
 
     @JDIAction("Is '{name}' checked")
     public boolean isChecked() {
-        return core().hasClass("Mui-checked");
+        return getCheckBoxes().hasClass("Mui-checked");
+    }
+
+    @JDIAction("'{name}' has {className} class")
+    public boolean hasClass(String className) {
+        return this.getCheckBoxes().hasClass(className);
```

Comment:
Is it possible to delete this method? You can try to use UIElement class instead of UIBaseElement in extends

Reply:
There is an error with is() method if we extend UIElement instead of UIBaseElement

Reply:
Please try to comment this method and see if tests pass or fail

Reply:
The problem is that we need custom asserts with timer, which are needed for stability of test. I think it will be better to keep UIBaseElement.